docs/integrations/sources/quickbooks.md

```diff
@@ -1,12 +1,76 @@
 # QuickBooks
 
-## Overview
+This page contains the setup guide and reference information for the QuickBooks Source connector.
 
-The QuickBooks source supports both Full Refresh and Incremental syncs. You can choose if this connector will copy only the new or updated data, or all rows in the tables and columns you set up for replication, every time a sync is run.
+## Prerequisites
 
-This source wraps the [Singer QuickBooks Tap](https://github.com/singer-io/tap-quickbooks).
+- Account at [Intuit Developer account](https://developer.intuit.com/app/developer/qbo/docs/get-started)
+- Start date
+- Realm ID
+- User Agent
+
+<!-- env:oss -->
+**For Airbyte Open Source:**
+
+- OAuth2.0 credentials (see [OAuth 2.0 playground](https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0-playground))
+<!-- /env:oss -->
+
+## Setup guide
+
+### Step 1: Set up QuickBooks
+
+Create a [Intuit Developer account](https://developer.intuit.com/app/developer/qbo/docs/get-started)
+
+<!-- env:oss -->
+**Airbyte Open Source additional setup steps**
+
+1. Create an application
+2. Obtain credentials. The easiest way to get these credentials is by using Quickbook's [OAuth 2.0 playground](https://developer.intuit.com/app/developer/qbo/docs/develop/authentication-and-authorization/oauth-2.0-playground)
+
+<!-- /env:oss -->
+
+<!-- env:cloud -->
+### Step 2: Set up the QuickBooks connector in Airbyte
+
+**For Airbyte Cloud:**
+
+1. [Log into your Airbyte Cloud](https://cloud.airbyte.com/workspaces) account.
+2. In the left navigation bar, click **Sources**. In the top-right corner, click **+ new source**.
+3. On the source setup page, select **QuickBooks** from the Source type dropdown and enter a name for this connector.
+4. Click `Authenticate your QuickBooks account`
+5. Log in and Authorize to the QuickBooks account.
+6. **Start date** - The date from which you'd like to replicate data for streams.
+7. **Realm ID** - The Labeled Company ID you'd like to replicate data for streams.
+8. **User Agent** - Process and email for API logging purposes.
+10. **Sandbox** - Turn on if you're going to replicate the data from the sandbox environment.
+11. Click **Set up source**.
+
+<!-- /env:cloud -->
+
+<!-- env:oss -->
+**For Airbyte Open Source:**
 
-### Output schema
+1. **Client ID** - The OAuth2.0 application ID
+2. **Client Secret** - The OAuth2.0 application secret
+3. **Refresh Token** - Refresh token used to get new access token every time the current one is expired
+4. **Access Token** - Access token to perform authenticated API calls with
```

Comment:
Please prefix optional inputs with `(Optional)` otherwise this is overwhelming for the user to read

Reply:
These fields are not optional, this [PR](https://github.com/airbytehq/airbyte/pull/20923) tells why